apps/comps/hooks/useCreateAgent.ts

```diff
@@ -13,21 +12,20 @@ export const useCreateAgent = () => {
   const queryClient = useQueryClient();
   const { trackEvent } = useAnalytics();
 
-  return useMutation({
-    mutationFn: async (data: CreateAgentRequest) => {
-      return apiClient.createAgent(data);
-    },
-    onSuccess: (response, variables) => {
-      trackEvent("UserSuccessfullyCreatedAgent", {
-        agent_id: response.agent?.id,
-        agent_name: variables.name,
-        agent_handle: variables.handle,
-      });
+  return useMutation(
+    tanstackClient.agent.createAgent.mutationOptions({
+      onSuccess: (response, variables) => {
+        trackEvent("UserSuccessfullyCreatedAgent", {
+          agent_id: response.agent.id,
+          agent_name: variables.name,
+          agent_handle: variables.handle,
+        });
 
-      // Invalidate relevant queries
-      queryClient.invalidateQueries({ queryKey: ["profile"] });
-      queryClient.invalidateQueries({ queryKey: ["agents"] });
-      queryClient.invalidateQueries({ queryKey: ["leaderboard"] });
-    },
-  });
+        // Invalidate relevant queries
+        queryClient.invalidateQueries({ queryKey: ["profile"] });
+        queryClient.invalidateQueries({ queryKey: ["agents"] });
+        queryClient.invalidateQueries({ queryKey: ["leaderboard"] });
```

Comment:
I think we really have to use the provided query key helper from the tanstackclient just for consistency.

Reply:
I think this must be fixed to approve.

Reply:
yeah so this is the part I wasn't sure about.  Can we update the keys for functions that haven't yet been migrated to the new rpc?  Like, the `leaderboard` query definitely hasn't been switched over yet.  I figured we'd go back and do a pass to update these once all the endpoints were migrated.

But looking again I guess the `profile` and `agents` ones at least could be updated, and I can leave leaderboard for now.  I just need to figure out if "profile" means user profile or agent profile...

Reply:
Okay updated, even caught a few previous rpc that had been migrated before my PR but still needed to be updated for invalidating their queries